Build failure: evhz

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

Add `pkgs.evhz` to your NixOS configuration, and rebuild the system.

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/317990006

### Relevant log output

```console
unpacking 1 channels...
building the system configuration...
evhz-unstable> Running phase: unpackPhase
evhz-unstable> unpacking source archive /nix/store/qnxff20rahk8gxnnasapibsm28yszqzj-source
evhz-unstable> source root is source
evhz-unstable> Running phase: patchPhase
evhz-unstable> Running phase: updateAutotoolsGnuConfigScriptsPhase
evhz-unstable> Running phase: configurePhase
evhz-unstable> no configure script, doing nothing
evhz-unstable> Running phase: buildPhase
evhz-unstable> evhz.c: In function 'main':
evhz-unstable> evhz.c:66:20: error: passing argument 2 of 'signal' from incompatible pointer type [-Wincompatible-pointer-types]
evhz-unstable>    66 |     signal(SIGINT, sigint);
evhz-unstable>       |                    ^~~~~~
evhz-unstable>       |                    |
evhz-unstable>       |                    void (*)(void)
evhz-unstable> In file included from evhz.c:18:
evhz-unstable> /nix/store/dj43clc5ff7jjnfmhbaj6q4q0h8kpfpm-glibc-2.40-66-dev/include/signal.h:88:57: note: expected '__sighandler_t' {aka 'void (*)(int)'} but argument is of type 'void (*)(void)'
evhz-unstable>    88 | extern __sighandler_t signal (int __sig, __sighandler_t __handler)
evhz-unstable>       |                                          ~~~~~~~~~~~~~~~^~~~~~~~~
evhz-unstable> evhz.c:36:6: note: 'sigint' declared here
evhz-unstable>    36 | void sigint() {
evhz-unstable>       |      ^~~~~~
evhz-unstable> /nix/store/dj43clc5ff7jjnfmhbaj6q4q0h8kpfpm-glibc-2.40-66-dev/include/signal.h:72:16: note: '__sighandler_t' declared here
evhz-unstable>    72 | typedef void (*__sighandler_t) (int);
evhz-unstable>       |                ^~~~~~~~~~~~~~
gnirehtet.apk> Running phase: unpackPhase
gnirehtet.apk> unpacking source archive /nix/store/1d699w4rz8zckvhnwnipij6m9jsnm540-source
gnirehtet.apk> source root is source
gnirehtet.apk> Running phase: patchPhase
gnirehtet.apk> Running phase: updateAutotoolsGnuConfigScriptsPhase
gnirehtet.apk> Running phase: configurePhase
gnirehtet.apk> no configure script, doing nothing
gnirehtet.apk> Running phase: buildPhase
gnirehtet.apk> no Makefile or custom buildPhase, doing nothing
gnirehtet.apk> Running phase: installPhase
gnirehtet.apk> Running phase: fixupPhase
gnirehtet.apk> shrinking RPATHs of ELF executables and libraries in /nix/store/47cpsa26j2cd0brh2xnsn51kvjyb79vb-gnirehtet.apk-2.5.1
gnirehtet.apk> checking for references to /build/ in /nix/store/47cpsa26j2cd0brh2xnsn51kvjyb79vb-gnirehtet.apk-2.5.1...
gnirehtet.apk> patching script interpreter paths in /nix/store/47cpsa26j2cd0brh2xnsn51kvjyb79vb-gnirehtet.apk-2.5.1
error: Cannot build '/nix/store/r0wsw9ij6g4mkag3n2qv077mgsyk65fw-evhz-unstable-2021-09-20.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/izsafjbj4dp7lzm0has7f0m35qsgkgxd-evhz-unstable-2021-09-20
error: Cannot build '/nix/store/g7vpkb8k3rm6d4lrgs7c7nd552zrhzlf-evhz-unstable-2021-09-20_fish-completions.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/byckdp01zcd9v6vcia78r9v8s7zy5nz4-evhz-unstable-2021-09-20_fish-completions
error: Cannot build '/nix/store/s6fmwk6zgi1i341zlh9dgg47f1vq60sc-system-path.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/ah3pwsfigrl1vdzhvjg8sdcfhvjza9bw-system-path
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.2-zen2, NixOS, 26.05 (Yarara), 26.05pre917670.c0b0e0fddf73`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+1`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/s9brl6fgs2n39hrdggj3144np5zdc8fv-nixos/nixos`

### Notify maintainers

@Tungsten842

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ev/evhz/package.nix
+++ b/pkgs/by-name/ev/evhz/package.nix
@@ -6,13 +6,13 @@
 
 stdenv.mkDerivation {
   pname = "evhz";
-  version = "unstable-2021-09-20";
+  version = "unstable-2025-10-09";
 
   src = fetchFromSourcehut {
     owner = "~iank";
     repo = "evhz";
-    rev = "35b7526e0655522bbdf92f6384f4e9dff74f38a0";
-    hash = "sha256-lC0CeN9YVhkSiooC59Dbom811jHvPDQcYl+KADUwVdQ=";
+    rev = "189e6c22a9acb01b6eea26659f6944e60f6cb3d8";
+    hash = "sha256-p2KH5wpw8PpsICutUnwrKmKPtQ/MauWb+DY55WIJPkE=";
   };
 
   buildPhase = "gcc -o evhz evhz.c";

